### PR TITLE
Update encrypt/decrypt API to add and remove padding (similar to C++ API for nacl)

### LINKED ISSUE #51: Uncaught panic inside enclave caused if the ciphertext is not padded with 0s correctly
We should remove the 16 byte padding from ciphertext before we return the result, as well as append 16 byte padding to any received ciphertext.

see: https://github.com/dchest/tweetnacl-js/issues/64
and: https://nacl.cr.yp.to/box.html

### PR DESCRIPTION
**Pull request by longtomjr**
Closes: #51

### CODE REVIEW

**Comment by PiDelport** [comment]
(Rebased for Makefile fix: #54)

File: rtc_tenclave/src/crypto.rs [hunk 1]
```diff
@@ -137,17 +185,34 @@ impl RtcCrypto for SodaBoxCrypto {
...
+            ]
+            .concat(),
             &nonce,
             their_pk,
             self.private_key.expose_secret(),
         ) {
-            Ok(_) => Ok(EncryptedMessage { ciphertext, nonce }),
+            Ok(_) => {
+                ciphertext.rotate_left(CRYPTO_BOX_BOXZEROBYTES);
+                ciphertext.truncate(CRYPTO_BOX_BOXZEROBYTES);
```
Comment (PiDelport) [inline rtc_tenclave/src/crypto.rs]: This should probably be the padded length minus the padding?

```suggestion
                ciphertext.truncate(ciphertext.len() - CRYPTO_BOX_BOXZEROBYTES);
```

File: rtc_tenclave/src/crypto.rs [hunk 2]
```diff
@@ -85,11 +104,21 @@ impl RtcCrypto for SodaBoxCrypto {
...
+
+        // Docs: https://nacl.cr.yp.to/box.html
+        //
+        // "The caller must ensure, before calling the crypto_box_open function,
+        // that the first crypto_box_BOXZEROBYTES bytes of the ciphertext c are all 0.
+        // The crypto_box_open function ensures (in case of success) that
+        // the first crypto_box_ZEROBYTES bytes of the plaintext m are all 0."
+        //
         match sodalite::box_open(
             &mut message,
```
Comment (PiDelport) [inline rtc_tenclave/src/crypto.rs:124]: The return result below still retains the padding for `message`… this should probably be stripped, for this PR?

**Review by PiDelport — CHANGES_REQUESTED** [review]

File: rtc_tenclave/src/crypto.rs (hunk 1 above)
Comment (longtomjr) [reply rtc_tenclave/src/crypto.rs]: Thanks for catching this!

File: rtc_tenclave/src/crypto.rs (hunk 2 above)
Comment (longtomjr) [reply rtc_tenclave/src/crypto.rs:124]: Thanks for catching this.

Comment (PiDelport) [reply rtc_tenclave/src/crypto.rs:124]: @longtomjr: I added some code to strip the prefix. 🚀

**Review by PiDelport — APPROVED** [review]
@longtomjr: Cool, feedback addressed! 🎉

Can you review the prefix stripping code I added?

**Review by Nghondzweni — APPROVED** [review]
Looking good.

**Comment by longtomjr** [comment]
@PiDelport @Nghondzweni. Please just have a peek at the last 2 commits where I added the tests, and then we can get this merged

**Comment by PiDelport** [comment]
(Refreshed)

**Review by PiDelport — APPROVED** [review]
Looks good! 🎉

I added some small readability tweaks to make the failure messages clearer about what's failing: I'll go ahead and merge.

**Pull request merged**